File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        size_t n = temperatures.size();
        if(n == 1) return {0};

        stack<pair<int,int>> st;
        vector<int> answer(n, 0);

        for(int i=n-1; i>=0; i--){
            while(!st.empty() && temperatures[i] >= st.top().first) st.pop();

            if(!st.empty()) answer[i] = st.top().second - i;
            st.push({temperatures[i], i});
        }

        return answer;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st,answer = list(), [0]*len(temperatures)

        for i,t in enumerate(temperatures):
            while st and t > temperatures[st[-1]]:
                prev_day = st.pop()
                answer[prev_day] = i - prev_day

            st.append(i)

        return answer